velox/type/DecimalUtil.h

```diff
@@ -54,8 +54,20 @@ class DecimalUtil {
     int128_t rescaledValue = inputValue.unscaledValue();
     auto scaleDifference = toScale - fromScale;
     if (scaleDifference >= 0) {
-      rescaledValue =
-          mul(rescaledValue, DecimalUtil::kPowersOfTen[scaleDifference]);
+      try {
+        rescaledValue = checkedMultiply<int128_t>(
```

Comment:
try-catch can introduce inefficiency. Let's use the `builtin` directly and check for overflow.

Reply:
we should also not throw for nullOnFailure.